traffic_portal/server.js

```diff
@@ -28,19 +28,29 @@ var constants = require('constants'),
 
 var config;
 
-try {
-    config = require('/etc/traffic_portal/conf/config');
-}
-catch(e) {
-    let file = "./conf/config";
-    if((process.env.NODE_ENV || "prod") === "dev")
-        file = './conf/configDev';
-    config = require(file);
+// forward-compatible in case this script becomes executable with a node
+// shebang.
+if (process.argv.length === 3 || process.argv.length === 4 && process.argv.slice(-2)[0] === "-c") {
+    config = require(process.argv.slice(-1)[0]);
+} else {
+    try {
+        config = require('/etc/traffic_portal/conf/config');
+    }
+    catch(e) {
+        let file = "./conf/config";
+        if((process.env.NODE_ENV || "prod") === "dev")
+            file = './conf/configDev';
+        config = require(file);
+    }
 }
 
-var logStream = fs.createWriteStream(config.log.stream, { flags: 'a' }),
-    useSSL = config.useSSL;
-
+const useSSL = config.useSSL;
+let logStream;
+if (!Object.prototype.hasOwnProperty.call(config, "log") || !config.log || !Object.prototype.hasOwnProperty.call(config.log, "stream") || !config.log.stream) {
+    logStream = null;
```

Comment:
I think we would want this to halt execution in this state.

Reply:
No, that's intentional, and causes logging to go to stdout per
> ... enables logging to `stdout` when logging is not configured in the configuration file.

Previously it *would* halt execution because it would be trying to either access a property of a "falsey" value or trying to open a writestream to something that isn't a valid filename. That's no longer desired behavior, which is why I do the intentional checks here.